caluma/caluma_workflow/domain_logic.py

```diff
@@ -268,6 +268,8 @@ def pre_cancel(case, validated_data, user, context=None):
 
     @staticmethod
     def post_cancel(case, user, context=None):
+        # TODO: remove in the next major release since `events.cancelled_case`
```

Comment:
Is there a convention for marking deprecations in the source code, something like `# DEPRECATED`? That way we could make it easier to find everything that needs cleaning up when we make a new major release.

Reply:
cc @winged @open-dynaMIX

Reply:
We usually emit `DeprecationWarning`s as can be seen [here](https://github.com/projectcaluma/caluma/blob/master/caluma/caluma_user/models.py#L17).

The process is also mentioned [here](https://github.com/projectcaluma/caluma/blob/master/docs/maintainers.md#release-management).

It's not so easy to implement here I guess, as we would need to emit the warning in the event receiver. Maybe it could be done [here](https://github.com/projectcaluma/caluma/blob/master/caluma/caluma_core/events.py#L8).

Reply:
Or we could try to emit the warning when registering a listener. We already use our own function for this, which currently is an alias for django's event register function.

If done this way, we could issue the warning on startup-time instead of runtime, which in my opinion would be cleaner. However it could still be the case that users go directly through django's `register()` instead, but currently that risk is rather low as we're the only users I'm aware of

Reply:
Great idea! The documented way of using caluma events is by using our `on()` decorator (currently just aliased), so we should be save even if we'd not be the only users. [This section](https://github.com/projectcaluma/caluma/blob/08879ce497ceef3b25ba54587360d5e8921e8d7a/docs/events.md#on) of the docs should be updated, if we do this.

Reply:
Ok done

Reply:
Nice work, exactly what I had in mind!